Autopilot/Path Manager/MPL3115A2.c

```diff
@@ -30,36 +33,80 @@ char initAltimeter() {
 #endif
     }
     return altimeterConnected;
+    // set pressure 
 }
 
-void calibrateAltimeter(long int pressure){
-    int msb = pressure >> 9;
-    int lsb = (pressure >> 1)& 255;
-    int offset = 0;
-    sendMessage(I2C_SLAVE_ADDRESS, BAR_IN_MSB, (char*)&msb, 1, WRITE);
-    sendMessage(I2C_SLAVE_ADDRESS, BAR_IN_LSB, (char*)&lsb, 1, WRITE);
-    sendMessage(I2C_SLAVE_ADDRESS, OFF_H, (char*)&offset, 1, WRITE);
+void calibrateAltimeter(int altitudeOffset){
+    if (altitudeOffset < -128 || altitudeOffset > 128){
+        error("Invalid altitude offset!");
+        // the range of altitude offset should be within -128 to 128
+    }
+    
+    else{
+        switchMode(1); //Barometer mode
+        float pressureFloat = 0.0;
+        //There's slight delay on pressure display, wait until pressure is not zero
+        while(pressureFloat <= 0.1){
```

Comment:
You'll get a faster response time if you do a delay instead. Generally floating point operations on an embedded system like ours are extremely slow without a dedicated floating point unit, which the picpilot doesnt have. The delay can be up to 50us, which is quite alot. We have delay functions available for use here (look in Timer.h or Clock.h?) . You can experiment with different delays and choose the smallest one that gives you an accurate value (or see if its available in the datasheet.